app/services/doc_auth/lexis_nexis/responses/true_id_response.rb

```diff
@@ -142,6 +142,23 @@ def billed?
           !!doc_auth_result
         end
 
+        def doc_auth_success?
+          transaction_status_passed? &&
+            true_id_product.present? &&
+            product_status_passed? &&
+            doc_auth_result_passed?
+        end
+
+        # @return [Boolean, nil]
+        # When selfie check is not required, return nil
+        # Otherwise:
+        #   return true if selfie check result == 'Pass'
+        #   return false
+        def selfie_success
+          return nil unless @liveness_checking_enabled
```

Comment:
```suggestion
          return nil if selfie_result.nil?
```
this value is eventually stored in the DocumentCaptureSessionResult ... should we return what's in the TrueID response instead of what we have configured? 🤔

Reply:
@amirbey , are you suggesting we return [nil, Pass, Fail] other than [nil, true, false]?

Reply:
@amirbey , ok here is one possible scenario, let's say
if we enabled selfie but the somehow response missing _selfie_result_ (maybe we misconfigured the workflow) , should _selfie_success_ be **nil** for **false**?

@eileen-nava  what do you think?

Reply:
i think it should reflect the TrueID Response which in this scenario would be nil ... looking at the system we'd  be able to determine that we didn't get a response and  troubleshoot accordingly

we have methods (ie: #all_passed) that will check if selfie was req'd along its value (passed/ailed/nil) for evaluating whether the TrueID response was successful

Reply:
@amirbey , the only issue I see is that #all_passed is not included as part of DocumentCaptureSessionResult.

Reply:
Re: the above suggested change, I think it makes sense to have what we store in the session reflect what the vendor returns.  (I realize I initially 👍🏻 the existing implementation.  I think the above reasoning is convincing for the change, but @dawei-nava, let me know if I am missing something.)

@dawei I remember in our group meeting Thursday, January 4th, you brought up a similar concern about a scenario where selfie is required but we don’t do the selfie check due to a misconfiguration.  @jonathanhooper from Ada said that if the selfie check is not performed and it was required, the user will have to do the workflow again.  He also said Ada has three checks downstream from our code to ensure that selfie check takes place.  Does that address your concern?

Reply:
@eileen-nava @amirbey  changed to return nil.